drivers/tee/optee/core.c

```diff
@@ -158,6 +158,12 @@ static void optee_get_version(struct tee_device *teedev,
 static int optee_open(struct tee_context *ctx)
 {
 	struct optee_context_data *ctxdata;
+	struct tee_device *teedev = ctx->teedev;
+	struct optee *optee = tee_get_drvdata(teedev);
+
+	if (teedev == optee->supp_teedev && optee->supp_cnt == 1)
+		return -EBUSY;
+	optee->supp_cnt++;
```

Comment:
Ah sorry we must increment the number of supplicants only if alloc is successful.